### PR TITLE
docs: Use apidoc directive in serialization-jackson.md

### PR DESCRIPTION
**Pull request by Captain1653**
References #22904

### CODE REVIEW

File: akka-docs/src/main/paradox/serialization-jackson.md [hunk 1]
```diff
@@ -176,11 +176,11 @@ Alternatively, you can define an intermediate trait for the case object and a cu
...
 
 On the other hand, if the ADT only has case objects, you can solve it by implementing a custom serialization for the enums. Annotate the `trait` with
-`@JsonSerialize` and `@JsonDeserialize` and implement the serialization with `StdSerializer` and
-`StdDeserializer`.
+[@JsonSerialize](https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/com/fasterxml/jackson/databind/annotation/JsonSerialize.html) and [@JsonDeserialize](https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/com/fasterxml/jackson/databind/annotation/JsonDeserialize.html) and implement the serialization with [StdSerializer](https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/com/fasterxml/jackson/databind/ser/std/StdSerializer.html) and
+[StdDeserializer](https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/com/fasterxml/jackson/databind/deser/std/StdDeserializer.html).
```
Comment (Captain1653) [inline akka-docs/src/main/paradox/serialization-jackson.md:183]: There is some link `JsonObject`  above. I'm not sure about the link.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

File: akka-docs/src/main/paradox/serialization-jackson.md [hunk 2]
```diff
@@ -68,16 +68,16 @@ For security reasons it is disallowed to bind the Jackson serializers to
...
+* @javadoc[java.io.Serializable](java.io.Serializable)
+* @javadoc[java.lang.Comparable](java.lang.Comparable).
 
 The deny list of possible serialization gadget classes defined by Jackson databind are checked
 and disallowed for deserialization.
 
 @@@ warning
 
-Don't use `@JsonTypeInfo(use = Id.CLASS)` or `ObjectMapper.enableDefaultTyping` since that is a security risk
+Don't use [@JsonTypeInfo(use = Id.CLASS)](https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-annotations/latest/com/fasterxml/jackson/annotation/JsonTypeInfo.html) or [ObjectMapper.enableDefaultTyping](https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/com/fasterxml/jackson/databind/ObjectMapper.html#enableDefaultTyping--) since that is a security risk
```
Comment (ennru) [inline akka-docs/src/main/paradox/serialization-jackson.md:80]: Even these may use the `@javadoc` directive if Jackson gets added to the `paradoxProperties` in https://github.com/akka/akka/blob/main/project/Paradox.scala#L32

Comment (Captain1653) [reply akka-docs/src/main/paradox/serialization-jackson.md:80]: @ennru Thank you for your advice.How can I do it? Something like that?:
`"javadoc.java.base_url" -> "https://docs.oracle.com/en/java/javase/11/docs/api/java.base/; SOME_OTHER_URL"`

Comment (ennru) [reply akka-docs/src/main/paradox/serialization-jackson.md:80]: The latter, the bit between `javadoc.` and `.base_url` must be a prefix of the package. E.g. `com.fasterxml.jackson`
This is how it is configured in Alpakka: https://github.com/akka/alpakka/blob/master/build.sbt#L392-L393

File: project/Paradox.scala [hunk 3]
```diff
@@ -34,6 +34,10 @@ object Paradox {
         "javadoc.akka.base_url" -> "https://doc.akka.io/japi/akka/2.6",
         "javadoc.akka.link_style" -> "direct",
         "javadoc.akka.http.base_url" -> "https://doc.akka.io/japi/akka-http/current",
+        "javadoc.com.fasterxml.jackson.annotation.base_url" -> "https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-annotations/latest/",
+        "javadoc.com.fasterxml.jackson.annotation.link_style" -> "direct",
+        "javadoc.com.fasterxml.jackson.databind.base_url" -> "https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/",
+        "javadoc.com.fasterxml.jackson.databind.link_style" -> "direct",
         "javadoc.akka.http.link_style" -> "frames",
```
Comment (ennru) [inline project/Paradox.scala:41]:
```suggestion
        "javadoc.akka.http.base_url" -> "https://doc.akka.io/japi/akka-http/current",
        "javadoc.akka.http.link_style" -> "frames",
        "javadoc.com.fasterxml.jackson.annotation.base_url" -> "https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-annotations/latest/",
        "javadoc.com.fasterxml.jackson.annotation.link_style" -> "direct",
        "javadoc.com.fasterxml.jackson.databind.base_url" -> "https://javadoc.io/doc/com.fasterxml.jackson.core/jackson-databind/latest/",
        "javadoc.com.fasterxml.jackson.databind.link_style" -> "direct",
```
Keep the `akka.http` properties together.

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by Captain1653** [comment]
Can it be added to 2.6.19 milestone? Please :)

**Pull request merged**